src/GafferScene/ShaderTweaks.cpp

```diff
@@ -316,8 +323,66 @@ bool ShaderTweaks::applyTweaks( IECoreScene::ShaderNetwork *shaderNetwork, Tweak
 				{
 					throw IECore::Exception( fmt::format( "Cannot apply tweak to \"{}\" : Mode must be \"Replace\" when inserting a connection", name ) );
 				}
+
 				const auto inputParameter = ShaderNetworkAlgo::addShaders( shaderNetwork, inputNetwork );
 				shaderNetwork->addConnection( { inputParameter, parameter } );
+
+				// TODO - it would be substantially more efficient to search for and process tweak sources
+				// just in `inputNetwork` before merging it to `shaderNetwork` ... but this would require
+				// dealing with weird connections where the input node handle is relative to `shaderNetwork`,
+				// but the output handle is relative to `inputNetwork`. This gets very confusing if there are
+				// nodes in the two networks with the same name, which get uniquified during addShaders.
+				// Doing this after merging simplifies all that.
+				std::vector<IECore::InternedString> shadersToDelete;
+				for( const auto &i : shaderNetwork->shaders() )
+				{
+					if( i.second->getName() == ShaderTweakProxy::shaderTweakProxyIdentifier() )
+					{
+						shadersToDelete.push_back( i.first );
+
+						ShaderNetwork::ConnectionRange range = shaderNetwork->outputConnections( i.first );
+						const std::vector<ShaderNetwork::Connection> outputConnections( range.begin(), range.end() );
+
+						for( const auto &c : outputConnections )
+						{
+							ShaderNetwork::Parameter dest = c.destination;
```

Comment:
Having `dest` as a mutable variable declared way before it is used is confusing. Just using `c.destination` everywhere would be clearer.

Reply:
Addressed